lib/test/t_CovarianceModel_std.cxx

```diff
@@ -122,39 +171,125 @@ int main(int, char *[])
     // Default input dimension parameter to evaluate the model
     const UnsignedInteger dimension = 2;
 
+    // 1) Squared exponential model
     {
       /* Default constructor */
       SquaredExponential myDefautModel;
-      fullprint << "myDefautModel = " << myDefautModel << std::endl;
+      test_model(myDefautModel);
 
       SquaredExponential myModel(dimension);
       test_model(myModel);
-
     }
+
+    // 2) Generalized exponential model
     {
       /* Default constructor */
       GeneralizedExponential myDefautModel;
-      fullprint << "myDefautModel = " << myDefautModel << std::endl;
+      test_model(myDefautModel);
 
       GeneralizedExponential myModel(Point(dimension, 10.0), 1.5);
       test_model(myModel);
     }
+
+    // 3) Absolute exponential model
     {
       /* Default constructor */
       AbsoluteExponential myDefautModel;
-      fullprint << "myDefautModel = " << myDefautModel << std::endl;
+      test_model(myDefautModel);
 
       AbsoluteExponential myModel(Point(dimension, 10.0));
       test_model(myModel);
     }
+
+    // 4) Matern  model
     {
       /* Default constructor */
       MaternModel myDefautModel;
-      fullprint << "myDefautModel = " << myDefautModel << std::endl;
+      test_model(myDefautModel);
 
       MaternModel myModel(Point(dimension, 8.0), 2.0);
       test_model(myModel);
     }
+
+    // 5) ExponentiallyDampedCosineModel
+    {
+
+      ExponentiallyDampedCosineModel myDefautModel;
+      test_model(myDefautModel);
+
+      const Point scale = {2, 2};
+      const Point amplitude = {3};
+
+      ExponentiallyDampedCosineModel myModel(scale, amplitude, 1);
+      assert_almost_equal(myModel.getScale(), scale, 1e-15, 1e-15);
+      assert_almost_equal(myModel.getAmplitude(), amplitude, 1e-15, 1e-15);
+      assert_almost_equal(myModel.getFrequency(), 1, 1e-15, 1e-15);
```

Comment:
I do not really understand the purpose of these tests.

However,  since this is about setting values, I would expect the equality to be exact.
```suggestion
      assert_almost_equal(myModel.getScale(), scale, 0.0, 0.0);
      assert_almost_equal(myModel.getAmplitude(), amplitude, 0.0, 0.0);
      assert_almost_equal(myModel.getFrequency(), 1, 0.0, 0.0);
```